### PR TITLE
LPD-87393 bulk delete confirmation modal

### PR DESCRIPTION
**Pull request by marcogalluzzi**
# What is this trying to solve?

https://liferay.atlassian.net/browse/LPD-87393

* The bulk deletion confirmation modal should look different when multiple spaces are selected for bulk deletion. See Figma.
* Also, I'm using the word "Item" instead "Entry" as discussed in Slack.

# How am I fixing it?

* Studying all different manners the bulk deletion can create the modal. 
* Fixing and simplifying all cases in typescript.
* Improvement: handle the case when select all is used in a view where one Space is being inspected.

Delete 1 item
delete 1 item

Delete multiple items
delete multiple items

Delete all items
delete all items

Delete multiple items from multiple spaces
delete multiple items from multiple spaces

Delete all items from multiple spaces
delete all items from multiple spaces

# How can you verify that it works?

See added tests.

### CODE REVIEW

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 1]
```diff
@@ -106,51 +130,50 @@ async function handleBulkDeletion({
...
+			spaces.every((space) => space.settings.trashEnabled);
+
+		someEntriesHaveTrashEnabled =
+			!spaces.length ||
+			spaces.some((space) => space.settings.trashEnabled);
+	}
+	else {
+		allEntriesHaveTrashEnabled = trashEnabled;
+		someEntriesHaveTrashEnabled = trashEnabled;
+	}
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: @gemini-code-assist

* When `trashEnabled` is provided and true the variable `allEntriesHaveTrashEnabled` will also be true, and therefore `showModal` is not called and we don't use the values of `confirmationMessage` and `title` returned by `bulkDeleteMessage`
* The same applies when `trashEnabled` is not provided

File: modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/ViewFolderSectionDisplayContext.java [hunk 2]
```diff
@@ -92,6 +95,22 @@ public Map<String, Object> getAdditionalProps() {
...
+
+				Group group = groupLocalService.fetchGroup(
+					objectEntryFolder.getGroupId());
+
+				if ((group != null) && _trashHelper.isTrashEnabled(group)) {
+					return true;
+				}
+
+				return false;
+			}
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/java/com/liferay/site/cms/site/initializer/internal/display/context/ViewFolderSectionDisplayContext.java:113]: It's a common pattern we use

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 3]
```diff
@@ -42,44 +42,68 @@ export function executeBulkDeleteAction(
  */
 function getBulkDeleteMessage(
 	selectedData: any,
-	allEntriesHaveTrashEnabled: boolean,
 	someEntriesHaveTrashEnabled: boolean
 ): {
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts:45]: Good point, but not needed

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts [hunk 4]
```diff
@@ -4,5 +4,9 @@
  */
 
 export function isFromRecycleBin(selectedData: any): boolean {
-	return selectedData.items[0].embedded.status.label === 'in-trash';
+	if (selectedData.items.length) {
+		return selectedData.items[0].embedded.status.label === 'in-trash';
+	}
```
[the remark this answers is not in the document]
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts]: not needed

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 5]
```diff
@@ -115,16 +115,24 @@ async function handleBulkDeletion({
...
-	const someEntriesHaveTrashEnabled = spaces.some(
-		(space) => space.settings.trashEnabled
-	);
+		allEntriesHaveTrashEnabled =
+			!!spaces.length &&
+			spaces.every((space) => space.settings.trashEnabled);
+
+		someEntriesHaveTrashEnabled =
+			!spaces.length ||
+			spaces.some((space) => space.settings.trashEnabled);
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: Wouldn't it be simpler to do `!!spaces?.every(...)`? Also, the `!spaces.length` for the "some" case is a typo, right? Shouldn't it be `!!`?

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 6]
```diff
@@ -115,16 +115,24 @@ async function handleBulkDeletion({
 	showConfirmationModal?: boolean;
 	trashEnabled?: boolean;
 }): Promise<void> {
-	const spaces = await getEntriesSpaces(selectedData?.items || []);
+	let allEntriesHaveTrashEnabled: boolean;
+	let someEntriesHaveTrashEnabled: boolean;
 
-	// Trash status checks
+	if (trashEnabled === null || trashEnabled === undefined) {
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: Simplify the condition to `trashEnabled ?? true`?

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts [hunk 7]
```diff
@@ -4,5 +4,9 @@
...
+
+	return false;
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts]: Wouldn't it be simpler to just use `return selectedData.items?.[0]?.embedded?.status?.label === 'in-trash'`?

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java [hunk 8]
```diff
@@ -0,0 +1,162 @@
...
+
+		Assert.assertEquals(Boolean.FALSE, additionalProps.get("trashEnabled"));
+
+		// No object entry folder in context
+
+		additionalProps = _getAdditionalProps(getMockHttpServletRequest());
+
+		Assert.assertFalse(
+			additionalProps.toString(),
+			additionalProps.containsKey("trashEnabled"));
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: Test blocks need to be sorted by comment.

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java [hunk 9]
```diff
@@ -0,0 +1,162 @@
...
+
+	@ClassRule
+	@Rule
+	public static final AggregateTestRule aggregateTestRule =
+		new AggregateTestRule(
+			new LiferayIntegrationTestRule(),
+			PermissionCheckerMethodTestRule.INSTANCE);
+
+	@Test
+	public void testGetAdditionalPropsTrashEnabled() throws Exception {
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: You need to name test methods the same as the method-under-test (`testGetAdditionalProps` in this case.)

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java [hunk 10]
```diff
@@ -0,0 +1,162 @@
...
+
+		DepotEntry depotEntry = _depotEntryLocalService.addDepotEntry(
+			Collections.singletonMap(
+				LocaleUtil.getDefault(), RandomTestUtil.randomString()),
+			null, DepotConstants.TYPE_SPACE,
+			new ServiceContext() {
+				{
+					setCompanyId(group.getCompanyId());
+					setUserId(TestPropsValues.getUserId());
+				}
```
Comment (adolfopa) [inline modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: Can you use ServiceContextTestUtil here?

**Review by adolfopa — CHANGES_REQUESTED** [review]

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts (hunk 6 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: My condition says `trashEnabled` is not defined at all, which happens when we are in multiple space views (All/Contents/Files). Your condition says `trashEnabled` is undefined or `true`.

Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: There is a shorthand that I can use if you prefer: `trashEnabled == null`
However it isn't used in any TSX file, this is why I didn't use it

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts (hunk 5 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: `every` returns `true` when the `spaces` array is empty. With `!!spaces.length` (which is the same as `spaces.length > 0`) I want to return true only when `every` returns true and the array is not empty. 

There is no typo with the `some` case. But I've refactored this case to better understand the conditions.

See last 2 commits to understand the readability improvements

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts (hunk 7 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/utils/isFromRecycleBin.ts]: You are right, it can be simplified with: `return selectedData.items[0]?.embedded?.status?.label === 'in-trash'`

Changed done in same commit.

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java (hunk 8 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: I tried to apply this rule. I don't know why I thought  "No object..." had to come last :facepalm:  Fixed now. Thanks.

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java (hunk 9 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: Done, in same commit. Thanks

File: modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java (hunk 10 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer-test/src/testIntegration/java/com/liferay/site/cms/site/initializer/internal/display/context/test/ViewFolderSectionDisplayContextTest.java]: Done is same commit. Thanks.

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts (hunk 6 above)
Comment (adolfopa) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts]: It is fine as is.

**Review by adolfopa — APPROVED** [review]

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 11]
```diff
@@ -106,42 +130,47 @@ async function handleBulkDeletion({
...
+			trashEnabled === null || trashEnabled === undefined;
+
+		const someEntriesHaveTrashEnabled = spaces.some(
+			(space) => space.settings.trashEnabled
+		);
+
+		const {confirmationMessage, title} = bulkDeleteMessage(
+			selectedData,
+			isMultipleSpacesView &&
+				(selectedData.selectAll || someEntriesHaveTrashEnabled)
```
Comment (bcarvalho89) [inline modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts:170]: Verified by deploying this PR locally and running a manual scenario: 2 items both in `Trash Space`, sent to the global Recycle Bin (`/web/cms/recycle-bin`), then selected together and bulk-deleted from the recycle bin.

**Before this PR (master):** modal title "Delete N Entries" with `delete-entries-confirmation` — the standard same-space confirmation, no multi-space copy.

**After this PR:** modal title "Delete 2 Items" with the multi-space warning including "Items from spaces with a Recycle Bin can be restored within 30 days." Two problems in the recycle-bin context:

1. The items are already in trash — permanently deleting from the recycle bin is final. There is no 30-day restore to advertise.
2. Both items came from `Trash Space` (one space). "From multiple spaces" is factually incorrect.

Root cause: `entriesFromMultipleSpacesSelected` evaluates true here because `trashEnabled` isn't passed in the recycle-bin view (so `isMultipleSpacesView=true`) and the items came from trash-enabled spaces (so `someEntriesHaveTrashEnabled=true`). The `isFromRecycleBin(selectedData)` check at line 152 forces the modal but doesn't influence the copy choice.

Suggested fix: when `isFromRecycleBin(selectedData)` is true, skip the "multiple spaces" branch and fall through to the regular `delete-x-items-confirmation` / `delete-item-confirmation` paths.

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts [hunk 12]
```diff
@@ -42,44 +42,68 @@ export function executeBulkDeleteAction(
...
+			confirmationMessage: getBodyHTML([
+				Liferay.Language.get(
+					'you-are-about-to-delete-the-selected-items-from-multiple-spaces'
+				),
+				Liferay.Language.get(
+					'bulk-delete-from-multiple-spaces-warning'
+				),
+				Liferay.Language.get('are-you-sure-you-want-to-continue'),
+			]),
+			title: selectedData.items.length
```
Comment (bcarvalho89) [inline modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts:60]: The PR description's Figma "Delete all items from multiple spaces" shows the title "Delete All Items". In practice, the `'delete-all-items'` branch only fires when `selectedData.items.length === 0`.

Verified live: 5 items across 2 spaces, Select All on `/cms/all`, then bulk Delete — title renders as "Delete 5 Items", not "Delete All Items", because FDS passes the visible items in `selectedData.items` alongside `selectAll: true`. The unit test at `deleteAssetEntriesBulkAction.test.ts:219` that covers the "Delete All Items" title sets `items: []`, which doesn't match the real FDS payload here.

If Figma fidelity matters, prefer `selectedData.selectAll` over `selectedData.items.length` for the title decision. Otherwise the title only matches Figma in the empty-items edge case.

Stylistic; not blocking.

**Review by bcarvalho89 — CHANGES_REQUESTED** [review]
Reviewed by deploying locally + manual exploration of the modal variants.

**Blocker — existing Playwright test regression.** `modules/test/playwright/tests/site-cms-site-initializer/main/bulkActions.spec.ts:125` (the `@LPD-62787` test "Can delete multiple contents across spaces with and without recycle bin enabled") asserts:

```ts
page.getByText('Some of the selected files')
```

That copy was the value of the removed `bulk-delete-cms-entries-confirmation` key. Verified failing after deploying this PR. The test was built specifically for the multi-space mixed-trash scenario this PR now renders with new copy — needs to be updated in this PR (e.g. assert "You are about to delete the selected items from multiple spaces").

**Recycle-bin modal copy regression** — see inline thread; pre-PR was the standard `delete-entries-confirmation`, after the PR it incorrectly shows the multi-space "30-day restore" warning even when all items came from one space.

**Verified good** — modal variants 1/2/3/4/5 from the PR description render per Figma in their intended scenarios. Unit tests pass (16/16).

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts (hunk 11 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts:170]: This has been fixed in this commit: [LPD-87393 Change the way we compute that the call comes from the Recy…](https://github.com/liferay-content-management/liferay-portal/pull/8244/commits/3cc2e6c281a5e600af6a121768ff4ccff0762bc6)

File: modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts (hunk 12 above)
Comment (marcogalluzzi) [reply modules/apps/site/site-cms-site-initializer/src/main/resources/META-INF/resources/js/main_view/props_transformer/actions/deleteAssetEntriesBulkAction.ts:60]: Not taken into account

**Review by bcarvalho89 — APPROVED** [review]
LGTM!
Thank you for the fixes and improvements.

**Review by adolfopa — APPROVED** [review]
